### PR TITLE
llmq/rpc: Adjust verify islock, don't err when transaction isn't known locally

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
This is a fix for platform, which is why I'm proposing to backport to 17

### CODE REVIEW

File: src/rpc/rpcquorums.cpp [hunk 1]
```diff
@@ -684,14 +684,10 @@ UniValue verifyislock(const JSONRPCRequest& request)
...
-                  ? "No such mempool or blockchain transaction"
-                  : "No such mempool transaction. Use -txindex to enable blockchain transaction queries";
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, errmsg);
-        }
-        if (!hash_block.IsNull()) {
-            pindexMined = LookupBlockIndex(hash_block);
+        if (GetTransaction(txid, tx, Params().GetConsensus(), hash_block, true)) {
+            if (!hash_block.IsNull()) {
+                pindexMined = LookupBlockIndex(hash_block);
+            }
```
Comment (UdjinM6) [inline src/rpc/rpcquorums.cpp:690]:
```suggestion
        if (GetTransaction(txid, tx, Params().GetConsensus(), hash_block, true) && !hash_block.IsNull()) {
            pindexMined = LookupBlockIndex(hash_block);
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by xdustinface — DISMISSED** [review]
utACK

**Comment by UdjinM6** [comment]
Let's also add some tests to make sure it works as expected ;) https://github.com/UdjinM6/dash/commit/39cc45e7cfec161b50fda5f447cc1d8a1211993f

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by xdustinface — APPROVED** [review]
utACK

**Pull request merged**